Automattic/vip-go-mu-plugins
# Jetpack Connection Pilot

## Description

Monitor Jetpack connections. Send alerts when there are connection issues.

Currently will only attempt a reconnection under one scenario: if the site was previously successfully connected and the domains haven't changed.

I can squash the commits when we're ready to merge.

## Steps to Test

1. Check out PR.
1. Pull the branch into the vip-go-mu-plugins on a sandbox
1. define the constant in `wp shell` : `define( 'WPCOM_VIP_RUN_CONNECTION_PILOT', true );`
1. Run `WPCOM_VIP_Jetpack_Connection_Pilot::run_connection_pilot()` in `wp shell`
1. Check the value of the `vip_jetpack_connection_pilot_healthcheck` option if it was successfully connected.
1. Otherwise, add a `return` statement [here](https://github.com/Automattic/vip-go-mu-plugins/blob/jetpack-connection-pilot/vip-jetpack/connection-pilot/class-jetpack-connection-pilot.php#L74) so you can see the error messages in the console.
